### ISSUE TITLE
In-place operators

### ISSUE THREAD

**Issue by vidartf** [hyperspy/hyperspy#833]
`Signal` currently implements the binary operators (+: `__add__`, etc.) but does not implement its in-place equivalents (+=: `__iadd__`). This isn't necessarily a problem, but if python doesn't find the in-place operators, it fakes them using the normal operators and then overwrites the original `Signal`.

```python
>>> s = hs.signals.Signal(np.ones((10,10)))
>>> s2 = s
>>> s += 3.4
>>> s is s2
False
```
This can lead to some confusion for the user, as line 3 seems to work as intended, but is not actually in-place. So, my proposal is that we should either create all the in-place operators and throw an error, or implement working copies of them.

**Comment by to266** [comment]
On one hand, I like / use the in-place operators, and adding them should be as trivial as adding the corresponding lines to https://github.com/hyperspy/hyperspy/blob/master/hyperspy/signal.py#L5096. However when working with out-of-memory (oom) signals, this thing would become impossible, so I would have to separate the behaviour anyway.... 

So my trouble is that I'd like to keep the short-hand notation, but not actually implement the in-place operators (at least for oom signals). I may just have to solve this problem when I actually encounter it, so it's likely not a big deal for now. We could, of course, start by documenting the current behaviour better (e.g. at least mention it somewhere).

**Comment by francisco-dlp** [comment]
Good point @vidartf. I think that you've found a subtle bug that requires fixing. I see @to266's point, but one step at a time...

**Comment by francisco-dlp** [comment]
Fixed in #867
